### PR TITLE
Use gmock after catkin

### PR DESCRIPTION
**Pull request by gaschler**
https://answers.ros.org/question/199680/catkin-doesnt-play-nice-with-google-mock/?answer=221147

### CODE REVIEW

File: cartographer_ros/CMakeLists.txt [hunk 1]
```diff
@@ -40,10 +40,15 @@ include("${CARTOGRAPHER_CMAKE_DIR}/functions.cmake")
...
 google_initialize_cartographer_project()
-google_enable_testing()
 
 find_package(catkin REQUIRED COMPONENTS ${PACKAGE_DEPENDENCIES})
 
+google_enable_testing()
+# Override Catkin's GTest configuration to use GMock.
+set(GTEST_FOUND TRUE)
+set(GTEST_INCLUDE_DIRS ${GMOCK_INCLUDE_DIRS})
+set(GTEST_LIBRARIES ${GMOCK_LIBRARIES})
```
Comment (dirk-thomas) [inline cartographer_ros/CMakeLists.txt]: I am not sure if this is still necessary. The latest version of `catkin` already does use GMock when it is available.

**Comment by ojura** [comment]
@gaschler this is not a good fix for #776. I submitted a proper fix, https://github.com/ros/catkin/pull/927, and it has been merged. I asked if they can make a release so we can get cartographer_ros CI unstuck.

**Comment by gaschler** [comment]
@ojura, I'm on it. My latest plan is to use cmake's `ExternalProject_Add` to build it reproducibly. We cannot wait for upstream changes.
(Abandoning this PR.)

**Comment by ojura** [comment]
Maybe consider just applying that patch in the Docker instance?

**Pull request closed without merging**